### PR TITLE
update docstring for expr.py, add test for Piecewise limit

### LINKED ISSUE #8804: series expansion of 1/x ignores order parameter
This should yield `1/x+O(x^8)`:

```
In [4]: series(1/x, n=8)
Out[4]: 
1
─
x
```

### LINKED ISSUE #18492: Limit of Piecewise function - NotImplementedError: Don't know how to calculate the mrv
Sympy 1.5.1

I'm playing around with Piecewise functions. Unfortunately, there are times where I receive this `NotImplementedError`. Here I provides a few examples:

```
import sympy as sp

x = sp.symbols("x", real=True)

def check(f, x, x0):
    m = sp.limit(f, x, x0, "-")
    p = sp.limit(f, x, x0, "+")
    v = f.subs(x, x0)
    return m == p and p == v

# Example 1
f31 = -sp.sqrt(sp.Abs(x - 3))
f32 = (x - 9) / 5
p3 = sp.Piecewise((f32, x < -1), (f32, x > 4), (f31, True))
check(p3, x, -1)

# Example 2
f41 = sp.sqrt(sp.Abs(x))
f42 = x / 4 + 1
p4 = sp.Piecewise((f41, x > 4), (f41, x < -4), (f42, True))
check(p4, x, 4)
```

Traceback:

```
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-120-c41dacdc8b53> in <module>
----> 1 check(p4, x, 4)

<ipython-input-85-f21ab98557af> in check(f, x, x0)
      1 def check(f, x, x0):
----> 2     m = sp.limit(f, x, x0, "-")
      3     p = sp.limit(f, x, x0, "+")
      4     v = f.subs(x, x0)
      5     return m == p and p == v

/usr/local/lib/python3.7/dist-packages/sympy/series/limits.py in limit(e, z, z0, dir)
     69     """
     70 
---> 71     return Limit(e, z, z0, dir).doit(deep=False)
     72 
     73 

/usr/local/lib/python3.7/dist-packages/sympy/series/limits.py in doit(self, **hints)
    259                             % (l, r))
    260             else:
--> 261                 r = gruntz(e, z, z0, dir)
    262             if r is S.NaN or l is S.NaN:
    263                 raise PoleError()

/usr/local/lib/python3.7/dist-packages/sympy/series/gruntz.py in gruntz(e, z, z0, dir)
    669 
    670     try:
--> 671         r = limitinf(e0, z)
    672     except ValueError:
    673         r = limitinf(e0, z, leadsimp=True)

/usr/local/lib/python3.7/dist-packages/sympy/core/cache.py in wrapper(*args, **kwargs)
     92             def wrapper(*args, **kwargs):
     93                 try:
---> 94                     retval = cfunc(*args, **kwargs)
     95                 except TypeError:
     96                     retval = func(*args, **kwargs)

/usr/local/lib/python3.7/dist-packages/sympy/series/gruntz.py in limitinf(e, x, leadsimp)
    433         e = e.subs(x, p)
    434         x = p
--> 435     c0, e0 = mrv_leadterm(e, x)
    436     sig = sign(e0, x)
    437     if sig == 1:

/usr/local/lib/python3.7/dist-packages/sympy/core/cache.py in wrapper(*args, **kwargs)
     92             def wrapper(*args, **kwargs):
     93                 try:
---> 94                     retval = cfunc(*args, **kwargs)
     95                 except TypeError:
     96                     retval = func(*args, **kwargs)

/usr/local/lib/python3.7/dist-packages/sympy/series/gruntz.py in mrv_leadterm(e, x)
    496         return (e, S.Zero)
    497     if Omega == SubsSet():
--> 498         Omega, exps = mrv(e, x)
    499     if not Omega:
    500         # e really does not depend on x after simplification

/usr/local/lib/python3.7/dist-packages/sympy/series/gruntz.py in mrv(e, x)
    295             return s, exp(expr)
    296     elif e.is_Function:
--> 297         l = [mrv(a, x) for a in e.args]
    298         l2 = [s for (s, _) in l if s != SubsSet()]
    299         if len(l2) != 1:

/usr/local/lib/python3.7/dist-packages/sympy/series/gruntz.py in <listcomp>(.0)
    295             return s, exp(expr)
    296     elif e.is_Function:
--> 297         l = [mrv(a, x) for a in e.args]
    298         l2 = [s for (s, _) in l if s != SubsSet()]
    299         if len(l2) != 1:

/usr/local/lib/python3.7/dist-packages/sympy/series/gruntz.py in mrv(e, x)
    309         return mrv(e.args[0], x)
    310     raise NotImplementedError(
--> 311         "Don't know how to calculate the mrv of '%s'" % e)
    312 
    313 

NotImplementedError: Don't know how to calculate the mrv of '(sqrt(Abs(4 - 1/_p)), _p < 1/8)'
```

Note that if I try to compute the same limits over the pieces, for example `sp.limit(f31, x, -1, "+")` everything works fine!

### PR DESCRIPTION
**Pull request by anutosh491**
#### Brief description of what is fixed or changed
Closes #18492
Closes #8804 
**This Pr consists of many minor fixes related to the series module**

1) Since the first time I used the series method I felt that the series docstring is quite suboptimal: it says that ` n is The number of terms upto which the series is to be expanded`, but this is only true if all terms would have a non-zero coefficient. For functions having terms which are 0 or rather terms which would be skipped (like sin, cos where I was genuinely confused when I first read the series documentation) , n represents the order of the expansion and not the number of terms in the output. I felt that this should be addressed . 

2) Added tests for piecewise functions which were failing earlier but now have been fixed elsewhere and work as expected on master

#### Other comments

This is my first time making a PR which is not trying to fix or solve an issue . And these two were pretty small changes to be made hence I have addressed them under a common PR and branch named "minor_fixes_for_series" . Not sure if this is good practice or not ! 

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/series/limitseq.py
Comment (0sidharth) [inline sympy/series/limitseq.py:95]: Why is this needed?

Comment (smichr) [reply sympy/series/limitseq.py:95]: Otherwise `dominant(n + Sum(k, (k, 0, n)), n)` wouldn't evaluate -- see OP. But I wonder if the input or None should just be returned if the expression didn't meet expectations. Not sure how be to introspect to know that, however. Perhaps if there are any objects with limits containing the series variable as a free symbol?

Comment (anutosh491) [reply sympy/series/limitseq.py:95]: I was stuck with an evaluation which was directly related to some issue or was inspired by some issue which involved a `Sum` expression inside `limit` . After some debugging I had realized that there's nothing wrong except this part which would be required to fix that . I'm forgetting exactly what was the question, I was trying to tackle ...... now that it's been some time since I made the Pr and add 2-3 weeks to that, when I tried that specific questions . I'll try to recall !

But anyways, I thought `doit`  would perform better in the general cases as in we have cases like this and `Sum` too returns None
```
>>> dominant(n**3 + n**2, n)
n**3
>>> dominant(n**3 + Product(n, (k, 1, 2)), n) 
RecursionError: maximum recursion depth exceeded while calling a Python object
```

**Comment by 0sidharth** [comment]
Looks good to me. Will merge in 24 hours if no issues

**Comment by anutosh491** [comment]
> Looks good to me. Will merge in 24 hours if no issues

Thanks @0sidharth

File: sympy/core/expr.py
Comment (smichr) [inline sympy/core/expr.py:2871]:
```suggestion
            up to which the series is to be expanded.
```

File: sympy/core/expr.py
Comment (smichr) [inline sympy/core/expr.py:3104]:
```suggestion
            up to which the series is to be expanded.
```

**Comment by smichr** [comment]
>dominant(n**2 + Sum(k, (k, 0, n)), n) # This returns None

I'm not sure that doit should be used to resolve this. Sometimes None means that the expression in its current form cannot be resolved. If the user wants to doit they should do so before using the function.

**Comment by 0sidharth** [comment]
+1. Though, some (possibly other) fix should still be implemented for the non-terminating examples

**Comment by anutosh491** [comment]
> +1. Though, some (possibly other) fix should still be implemented for the non-terminating examples

Thanks for the suggestions ! I will look for better ways to approach this and make relevant changes

**Comment by anutosh491** [comment]
okay so the point due to which something like `dominant(n + Product(n, (k, 1, 2)), n)` doesn't work is that `limit_seq(Product(n, (k, 1, 2)) / n, n) ` leads to infinite recursion and maybe would require a better fix than what this Pr was opened for , which was to address couple minor issues I could see in code ! and maybe `doit()` would do the job but might not be an intended fix . I'll be opening another Pr and try to fix the limit_seq bug , untill then I'm satisfied with the Pr and can be merged after review

**Comment by anutosh491** [comment]
This could go in now that we plan to deal with the fix for `limit_seq(Product(n, (k, 1, 2)) / n, n) ` separately ! Would like to know your views too on the changes made after my latest commit/ the message above

**Comment by smichr** [comment]
So this is now just a PR for fixing typos and adding a test. Could you please edit the OP? I would do so but you might want the text for a different PR.

**Review by smichr — APPROVED** [review]

**Comment by anutosh491** [comment]
Ready to go in . Thanks for the reviews and help !!

**Comment by anutosh491** [comment]
@smichr

**Comment by anutosh491** [comment]
ping @smichr for review !

**Comment by anutosh491** [comment]
ping @smichr for merging

**Comment by anutosh491** [comment]
cc @smichr for merging !

**Comment by anutosh491** [comment]
ping @smichr I think this is good to go !

**Comment by anutosh491** [comment]
ping @smichr this looks ready to go in !

**Comment by anutosh491** [comment]
I feel this is good to go ! if no changes are required to be made here could this be merged ?

**Comment by anutosh491** [comment]
CC: @oscarbenjamin , could you help me with this Pr , I know you these you don't deal with the `series` module but this is just a basic pr addressing some tests and docstrings . The changes have been **approved** long back but haven't heard anything substantial since then! I feel this good to go but maybe you could review it once ?

**Comment by anutosh491** [comment]
@oscarbenjamin I would be glad if you could help me with the authors check here .The branch had merge conflicts due to changes in imports , which I had tried to fix (through the second last commit )

1) But when i ran the `mailmap_check.py` file for checking for any reordering , it reordered id's of other authors too which was weird , but i made the commit thinking it might be the correct step but wasn't ! 
2) Then I took the mailmap file from a pr I made yesterday (which had passed the authors check) and committed the same file here , but here it fails which again is weird . mailmap file confuses me sometimes :( 
It shows the message and I have no clue why an error regarding Akshansh's id shows up here !

```
Ambiguous names in .mailmap
This email address appears for multiple entries:
Person: Akshānsh <akshansh@tuta.io>
Mailmap entries:
Akshansh Bhatt <akshansh@tuta.io> Akshansh <53227[12](https://github.com/sympy/sympy/runs/5376803817?check_suite_focus=true#step:6:12)7+akshanshbhatt@users.noreply.github.com>
Akshansh Bhatt <akshansh@tuta.io> akshanshbhatt <53227127+akshanshbhatt@users.noreply.github.com>
```

**Comment by oscarbenjamin** [comment]
> 2\. Then I took the mailmap file from a pr I made yesterday (which had passed the authors check)

You can't just copy a file across branches like that because git keeps track of changes one commit at a time. Look at the "files changed" tab in this PR and revert all of your changes to the .mailmap file.

The lines you have added are:
```
anutosh491 <87052487+anutosh491@users.noreply.github.com>
anutosh491 <andersonbhat491@gmail.com> Anutosh Bhat <87052487+anutosh491@users.noreply.github.com>
```
Each line should begin with the name and email address that you want to have in the AUTHORS file. Do you want your name to be recorded as "anutosh491" rather than "Anutosh Bhat"? What email address do you want to be recorded with that?

**Comment by anutosh491** [comment]
> You can't just copy a file across branches like that because git keeps track of changes one commit at a time. Look at the "files changed" tab in this PR and revert all of your changes to the .mailmap file.


Thanks for the help here @oscarbenjamin . Don't know why I thought that committing a working mailmap file was the best way to fix the issue I faced , that was surely foolish on my part ! .
Now that the mailmap issue has been fixed, I would be glad if you could help me get this pr in . @smichr had approved the changes on DEC 11 and from since then I have been trying to get my fellow developers/ maintainers to notice the pr but didn't succeed ? 
It's a `collection of tests and docstring` ( because I realized there were many issues which just needed tests and docstring changes in the series module at one point , hence didn't create different prs as everything was almost basic and to avoid opening too many prs at once). So I would be glad if you could help me here . Thanks for the above help anyways 👍

File: .mailmap
Comment (oscarbenjamin) [inline .mailmap:1242]: This is still not correct because it is listing you as two separate contributors.

Comment (anutosh491) [reply .mailmap:1242]: I tried removing the second option . The git diff I applied
```
--- a/.mailmap
+++ b/.mailmap
@@ -1238,8 +1238,7 @@ alijosephine <alijosephine@gmail.com>
 andreo <andrey.torba@gmail.com>
-anutosh491 <andersonbhat491@gmail.com> Anutosh Bhat <87052487+anutosh491@users.noreply.github.com>
+anutosh491 <andersonbhat491@gmail.com> anutosh491 <87052487+anutosh491@users.noreply.github.com>
 arooshiverma <av22@iitbbs.ac.in>
```
After running `mailmap_check.py` , I got this message which I felt was not what we would want !
```
←[31mThis author is not included in the .mailmap file:←[0m
Anutosh Bhat <87052487+anutosh491@users.noreply.github.com>
←[31m
The .mailmap file needs to be updated because there are commits with
unrecognised author/email metadata.←[0m
←[31m
For instructions on updating the .mailmap file see:
https://github.com/sympy/sympy/wiki/Development-workflow#update-
mailmap←[0m
←[32m
The following authors will be added to the AUTHORS file at the time of
the next SymPy release.←[0m

        Anutosh Bhat <87052487+anutosh491@users.noreply.github.com>
        Siddhant Jain <siddhantashoknagar@gmail.com>
        Tirthankar Mazumder <63574588+wermos@users.noreply.github.com>
```

Comment (anutosh491) [reply .mailmap:1242]: And if I get rid of the first option using this 
```
index 308a6e7526..d80b9165a3 100644
--- a/.mailmap
+++ b/.mailmap
@@ -1238,8 +1238,7 @@ alijosephine <alijosephine@gmail.com>
 amsuhane <ayushsuhane99@iitkgp.ac.in> amsuhane <“ayushsuhane99@iitkgp.ac.in”>
-anutosh491 <andersonbhat491@gmail.com> Anutosh Bhat <87052487+anutosh491@users.noreply.github.com>
+Anutosh Bhat <87052487+anutosh491@users.noreply.github.com>
 arooshiverma <av22@iitbbs.ac.in>
```
I got this message, which again seems wrong !
```
←[31mThis author is not included in the .mailmap file:←[0m
anutosh491 <andersonbhat491@gmail.com>
←[31m
The .mailmap file needs to be updated because there are commits with
unrecognised author/email metadata.←[0m
←[31mThe mailmap file was reordered←[0m
←[31m
For instructions on updating the .mailmap file see:
https://github.com/sympy/sympy/wiki/Development-workflow#update-
mailmap←[0m
←[32m

```

Comment (oscarbenjamin) [reply .mailmap:1242]: You are already in the AUTHORS file under this name and email address:
https://github.com/sympy/sympy/blob/0bc2587aef09ab150bd408b7cd9e08baf7b5fd9c/AUTHORS#L1112
Is that the name and email address that you want to have in the AUTHORS file? (I asked this question above and you didn't answer)

Comment (oscarbenjamin) [reply .mailmap:1242]: Then the change that needs to be made is this:
```diff
diff --git a/.mailmap b/.mailmap
index b39173b..52d0c68 100644
--- a/.mailmap
+++ b/.mailmap
@@ -1239,6 +1239,7 @@ amsuhane <ayushsuhane99@iitkgp.ac.in> amsuhane <“ayushsuhane99@iitkgp.ac.in”
 anca-mc <anca-mc@users.noreply.github.com>
 andreo <andrey.torba@gmail.com>
 anutosh491 <andersonbhat491@gmail.com> Anutosh Bhat <87052487+anutosh491@users.noreply.github.com>
+anutosh491 <andersonbhat491@gmail.com> anutosh491 <87052487+anutosh491@users.noreply.github.com>
 arooshiverma <av22@iitbbs.ac.in>
 bluebrook <perl4logic@gmail.com>
 carstimon <carstimon@gmail.com>
```
Each line begins with the name and email address that is in the AUTHORS file and then lists the other name and email address that has been used for some of your commits.

Comment (anutosh491) [reply .mailmap:1241]: Thanks for the help again ! Can't thank you enough , mailmap has been confusing me since a while now , but now I know how to deal with it in other prs too , i'll handle other too prs which need such fix like this one for consistency

**Comment by anutosh491** [comment]
@oscarbenjamin does this look good to you now ?

File: .mailmap
Comment (oscarbenjamin) [inline .mailmap:1242]: This is now correct.

Comment (anutosh491) [reply .mailmap:1242]: Thanks for your help, You could help me with the pr too ? I am quite confident of the changes here !

**Comment by eagleoflqj** [comment]
I squashed the commits since the intermediate state is not useful. Also removed the release note.
It looks good to me. I'll just wait for other suggestions, and then merge it.

**Comment by anutosh491** [comment]
Thanks @eagleoflqj , I am quite happy with the changes here . Didn't have to do too much here as everything was simple and straightforward.

**Pull request merged**
